src/pages-and-resources/discussions/app-list/AppList.test.jsx

```diff
@@ -20,84 +19,109 @@ import { fetchDiscussionSettings, fetchProviders } from '../data/thunks';
-    expect(queryAllByRole(container, 'radio')).toHaveLength(appCount);
-  });
+    test('hides the FeaturesList at desktop sizes', async () => {
+      renderComponent();
+      expect(screen.queryByText(messages['supportedFeatureList-mobile-show'].defaultMessage)).not.toBeInTheDocument();
+    });
 
-  test('displays the FeaturesTable at desktop sizes', async () => {
-    await mockStore(piazzaApiResponse);
-    expect(queryByRole(container, 'table')).toBeInTheDocument();
-  });
+    test('displays the FeaturesList at mobile sizes', async () => {
+      renderComponent(breakpoints.extraSmall.maxWidth);
+      const appCount = store.getState().discussions.appIds.length;
+      expect(screen.queryAllByText(messages['supportedFeatureList-mobile-show'].defaultMessage)).toHaveLength(appCount);
+    });
 
-  test('hides the FeaturesTable at mobile sizes', async () => {
-    await mockStore(piazzaApiResponse, breakpoints.extraSmall.maxWidth);
-    expect(queryByRole(container, 'table')).not.toBeInTheDocument();
+    test('selectApp is called when an app is clicked', async () => {
+      renderComponent();
+      userEvent.click(screen.getByLabelText('Select Piazza'));
+      const clickedCard = screen.getByRole('radio', { checked: true });
+      expect(within(clickedCard).queryByLabelText('Select Piazza')).toBeInTheDocument();
+    });
   });
 
-  test('hides the FeaturesList at desktop sizes', async () => {
-    await mockStore(piazzaApiResponse);
-    expect(queryByText(container, messages['supportedFeatureList-mobile-show'].defaultMessage)).not.toBeInTheDocument();
-  });
+  describe('AppList for Non Admin role', () => {
+    beforeEach(async () => {
+      initializeMockApp({
+        authenticatedUser: {
+          userId: 3,
+          username: 'abc123',
+          administrator: false,
+          roles: [],
+        },
+      });
 
-  test('displays the FeaturesList at mobile sizes', async () => {
-    await mockStore(piazzaApiResponse, breakpoints.extraSmall.maxWidth);
-    const appCount = store.getState().discussions.appIds.length;
-    expect(queryAllByText(container, messages['supportedFeatureList-mobile-show'].defaultMessage)).toHaveLength(appCount);
-  });
+      store = await initializeStore();
+      axiosMock = new MockAdapter(getAuthenticatedHttpClient());
+      await mockStore(legacyApiResponse, 'legacy');
+    });
 
-  test('selectApp is called when an app is clicked', async () => {
-    await mockStore(piazzaApiResponse);
-    userEvent.click(getByLabelText(container, 'Select Piazza'));
-    const clickedCard = getByRole(container, 'radio', { checked: true });
-    expect(queryByLabelText(clickedCard, 'Select Piazza')).toBeInTheDocument();
+    test('does not display two edx providers card for non admin role', async () => {
+      renderComponent();
+      const appCount = store.getState().discussions.appIds.length;
+      expect(queryAllByRole(container, 'radio')).toHaveLength(appCount - 1);
```

Comment:
You can directly pass the length instead of creating a variable appCount. It's recommended to avoid declaring variables that are only used once for the sake of cleaner and more concise code.